fvm/storage/state/transaction_state.go

```diff
@@ -187,6 +187,21 @@ func NewTransactionState(
 	}
 }
 
+// NewTransactionStateFromExecutionState constructs a new state transaction directly
+// from an execution state.
+func NewTransactionStateFromExecutionState(
```

Comment:
Maybe call this from `NewExecutionState`, the two functions mostly do the same thing